airflow/logging_config.py

```diff
@@ -13,52 +13,38 @@
 # limitations under the License.
 #
 import logging
-import os
-import sys
-
 from logging.config import dictConfig
 from airflow import configuration as conf
 
-log = logging.getLogger(__name__)
-
+from airflow.utils.module_loading import import_string
 
-def configure_logging(custom_config_path=None):
-    if custom_config_path is not None:
-        config_path = custom_config_path
-    else:
-        config_path = conf.get('core', 'logging_config_path')
-
-    try:
-        path = os.path.dirname(config_path)
-        sys.path.append(path)
-
-        filename = os.path.basename(config_path).replace('.py', '')
+log = logging.getLogger(__name__)
 
-        logging_module = __import__(filename)
-        logging_config = logging_module.LOGGING_CONFIG
 
-        # Make sure that the variable is in scope
-        assert (isinstance(logging_config, dict))
+def configure_logging():
+    logging_class_path = conf.get("core", "logging_config_class")
 
-        log.info(
-            'Successfully imported user-defined logging config from %s',
-            config_path
-        )
-    except AttributeError:
-        # Import default logging configurations.
-        log.warn(
-            'Loaded custom config from %s, but could not find LOGGING_CONFIG, Using default config instead',
-            config_path
-        )
-        from airflow.config_templates.default_airflow_local_settings import \
-            DEFAULT_LOGGING_CONFIG as logging_config
-    except Exception:
-        # Import default logging configurations.
-        log.debug(
-            'Unable to load custom logging, using default config instead'
-        )
-        from airflow.config_templates.default_airflow_local_settings import \
+    if logging_class_path == '':
+        from airflow.config_templates.airflow_local_settings import \
             DEFAULT_LOGGING_CONFIG as logging_config
+        log.debug('Unable to load custom logging, using default config instead')
+    else:
+        try:
+            logging_config = import_string(logging_class_path)
+
+            # Make sure that the variable is in scope
+            assert (isinstance(logging_config, dict))
+
+            log.info(
```

Comment:
Use .format

Reply:
We use this way of formatting the logging messages throughout Airflow:
https://github.com/apache/incubator-airflow/pull/2592